data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -148,6 +149,21 @@ public void shutdown() {
 
     }
 
+    @Override
+    public Collection<Record<Event>> applicableEventsForPeerForwarding(Collection<Record<Event>> records) {
+        if (whenCondition == null) {
+            return records;
+        }
+        final Collection<Record<Event>> recordsOut = new ArrayList<>();
+        for (Record<Event> record: records) {
+            Event event = record.getData();
+            if (expressionEvaluator.evaluateConditional(whenCondition, event)) {
```

Comment:
Let's share the same logic here as we use in the `execute` method to ensure they remain consistent.

Maybe you could make a method that evaluates the conditional in both.

```
private boolean isEventApplicable(Record<Event> record)
```